stl/inc/atomic

```diff
@@ -628,6 +729,33 @@ struct _Atomic_storage<_Ty, 4> { // lock-free using 4-byte intrinsics
         return false;
     }
 
+#if _HAS_CXX20
+
+    void wait(_Ty _Expected, const memory_order _Order = memory_order_seq_cst) const noexcept {
+        long _Spin_context = 0;
+        for (;;) {
+            const long _Expected_bytes = _Atomic_reinterpret_as<long>(_Expected);

```

Comment:
Out of curiosity, does it make sense to use `long` here which theoretically has different sizes or should it be type with a definitive size such as `uint32_t`? 

Same for `long long`

Reply:
Probably question not to me, I use an approach consistent with `compare_exchange_strong`.
I used long for `_Spin_context` also for consistency.

My guess on the rationale is to avoid `<cstdint>` include. Like, you want `std::atomic_uint64_t`, you get it, but if you also want non-atomic `std::uin64_t`, you are forced to include `<cstdint>` as well.

Anyway, in Visual Studio `long` has always been 4 bytes long so far.